pkg/R/utils.R

```diff
@@ -218,3 +218,53 @@ sortKeyValueList <- function(kv_list) {
   keys <- sapply(kv_list, function(x) x[[1]])
   kv_list[order(keys)]
 }
+
+## Utility function to generate compact R lists from grouped rdd 
+#  Used in Join-family functions
+genCompactLists <- function(tagged_list, cnull) {
+  len <- length(tagged_list)
+  num <- length(cnull)
+  lists <- list(vector("list", len), vector("list", len))
+  index <- list(1, 1)
+
+  for (x in tagged_list) {
+    tag <- x[[1]]
+    idx <- index[[tag]]
+    lists[[tag]][[idx]] <- x[[2]]
+    index[[tag]] <- idx + 1
+  }
+
+  len <- lapply(index, function(x) x-1)
+  for (i in (1:num)) {
```

Comment:
Basically I made this function more general to generate multiple compact lists each with specified tag(`1L`, `2L` ...), it may be useful in the future.